packages/std/src/metadata.rs

```diff
@@ -21,5 +22,292 @@ pub struct DenomMetadata {
+        // Test normal case with provided denom_units
+        let json_with_units = json!({
+            "description": "Test Token",
+            "denom_units": [
+                {
+                    "denom": "utest",
+                    "exponent": 6,
+                    "aliases": ["microtest"]
+                }
+            ],
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata_with_units: DenomMetadata = serde_json::from_value(json_with_units).unwrap();
+        assert_eq!(metadata_with_units.denom_units.len(), 1);
+        assert_eq!(metadata_with_units.denom_units[0].denom, "utest");
+
+        // Test with null aliases inside denom_units - should deserialize as empty vec
+        let json_with_null_aliases = json!({
+            "description": "Test Token",
+            "denom_units": [
+                {
+                    "denom": "utest",
+                    "exponent": 6,
+                    "aliases": null
+                }
+            ],
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata_with_null_aliases: DenomMetadata =
+            serde_json::from_value(json_with_null_aliases).unwrap();
+        assert_eq!(metadata_with_null_aliases.denom_units.len(), 1);
+        assert_eq!(
+            metadata_with_null_aliases.denom_units[0].aliases,
+            Vec::<String>::new()
+        );
+    }
+
+    #[test]
+    fn deserialize_denom_metadata_with_missing_fields_fails() {
+        // Missing denom_units should be treated like null
+        let json_missing_denom_units = json!({
+            "description": "Test Token",
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata: Result<DenomMetadata, Error> =
+            serde_json::from_value(json_missing_denom_units);
+        assert!(metadata.is_err());
+
+        let json_missing_alias = json!({
+            "description": "Test Token",
+            "base": "utest",
+            "denom_units": [
+                {
+                    "denom": "utest",
+                    "exponent": 6,
+                }
+            ],
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata_missing_alias: Result<DenomMetadata, Error> =
+            serde_json::from_value(json_missing_alias);
+        assert!(metadata_missing_alias.is_err());
+    }
+}
+
+#[cfg(test)]
+mod integration_tests {
```

Comment:
I don't really see how these are integration tests. They are unit tests just like the other ones.
Could you just combine the tests in here with the ones in `unit_tests` into a single `tests` module here in this file?

Reply:
Sure, no problem. I pushed codes with this change.

Reply:
Thanks for reviewing my codes!